File: contest/CF_1181/pB.py
l = int(input())
n = input()



first = l//2 + 1
second = l//2
while second < l and n[second] == '0':
    second += 1

while n[first] == '0':
    first -= 1

ans = -1
if second == l:
    ans = int(n[0:first]) + int(n[first:l])
elif first == 0:
    ans = int(n[0:second]) + int(n[second:l])
else:
    ans = min(int(n[0:second]) + int(n[second:l]),int(n[0:first]) + int(n[first:l]))

print(ans)

File: 0contest/kotlin_hero_3/pB.kt
import java.util.*
private fun readLn() = readLine()!! // string line
private fun readInt() = readLn().toInt() // single int
private fun readStrings() = readLn().split(" ") // list of strings
private fun readInts() = readStrings().map { it.toInt() } // list of ints

fun main() {
    var t = readInt();

    for (ul in 1..t) {
        var n = readInt();
        var seg = arrayListOf<Pair<Int,Int>>();
        var can = arrayListOf<Int>();

        for (i in 1..n) {
            var (u, v) = readInts();
            seg.add(Pair(u, v));
            can.add(u);
            can.add(u-1);
            can.add(v);
            can.add(v+1);
        }

        var ans = -1;

        for (v in can) {
            var cnt = 0;
            for (p in seg) {
                if (v >= p.first && v <= p.second) {
                    cnt++;
                }
            }
            if (cnt == 1) {
                ans = v;
            }
        }

        println(ans);
    }
}